build-system/compile/compile.js

```diff
@@ -153,7 +153,7 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
 
     // Add babel plugin to remove unwanted polyfills in esm build
     if (options.esmPassCompilation) {
-      compilationOptions['dest'] = './dist/esm/';
+      compilationOptions['dest'] = `./${OUTPUT_DIR}esm/`;
```

Comment:
Isn't the build pipeline supposed to be windows compatible?
If yes, please use `path.join`